include/godot_cpp/core/class_db.hpp

```diff
@@ -112,6 +112,33 @@ class ClassDB {
 	template <class T, bool is_abstract>
 	static void _register_class(bool p_virtual = false, bool p_exposed = true);
 
+	template <class T>
+	static GDExtensionObjectPtr create(void *data) {
+		if constexpr (!std::is_abstract_v<T>) {
+			T *new_object = memnew(T);
+			return new_object->_owner;
+		} else {
+			return nullptr;
+		}
+	}
+
+	template <class T>
+	static GDExtensionClassInstancePtr recreate(void *data, GDExtensionObjectPtr obj) {
```

Comment:
Similar to the above, maybe:

```suggestion
	static GDExtensionClassInstancePtr _recreate_instance_func(void *data, GDExtensionObjectPtr obj) {
```